src/components/chatroom/ChatBox.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+
+import MyChat from './MyChat';
+import OthersChat from './OthersChat';
+
+const ChatBox = ({ chatList, setChatList, users }) => {
+  const scrollRef = useRef();
+
+  useEffect(() => {
+    scrollRef.current.scrollBy({
+      top: 1000,
+      behavior: 'smooth',
+    });
+  }, [chatList]);
+
+  return (
+    <ChatList ref={scrollRef}>
+      {chatList.map((chat) =>
+        chat.userId === 0 ? (
+          <MyChat key={chat.date} chat={chat} users={users} />
+        ) : (
+          <OthersChat
+            key={chat.date}
+            chat={chat}
+            chatList={chatList}
+            setChatList={setChatList}
+            users={users}
+          />
+        )
+      )}
+    </ChatList>
```

Comment:
map() 함수에 두 번째 인자로 index를 넣어주면, map 내에서 자체적으로 인덱싱 순서를 받아서 쓸 수 있어요!

```suggestion
    <ChatList ref={scrollRef}>
      {chatList.map((chat,index) =>
        chat.userId === 0 ? (
          <MyChat key={index} chat={chat} users={users} />
        ) : (
          <OthersChat
            key={index}
            chat={chat}
            chatList={chatList}
            setChatList={setChatList}
            users={users}
          />
        )
      )}
    </ChatList>
```
이런식으로요!
`date`가 unique한 값이라서 key로 사용해도 문제 없겠지만, 추후에 해당 index로 작업하는 편이 저는 더 직관적인 것 같아서 제안드립니다!

Reply:
저도 처음에 그 방식을 사용했었는데, 저번 주차 PR에 선종 님께서 해당 방식은 좋지 않다고 코멘트 남겨 주셨습니다...! 그때 남겨 주셨던 코멘트예요! ⬇️

map의 index를 컴포넌트의 key로 사용하는것은 바람직하지 않습니다. key를 사용하는 이유가 불필요한 리렌더링의 방지인데, 컴포넌트가 리렌더링 될 때마다 map이 실행되면서 <ListItemWrapper>의 key가 매번 다시 설정되면서 결국 모든 컴포넌트가 리렌더링 됩니다.

Reply:
와우 배워갑니다!!